### ISSUE TITLE
Using the Nix setup with low disk space

### ISSUE THREAD

**Issue by sjakobi** [dhall-lang/dhall-haskell#1823]
Whenever I try using the Nix setup, it seems to download and build close to 20 GB of artifacts, which tends to fill the partition for my root directory to a dangerous level. Obviously I should finally set up a larger partition, but I'm also wondering whether there's a way to use the Nix setup without pulling in _all_ the dependencies, for example by building only with GHC-8.4.4. Is that possible?

**Comment by Gabriel439** [comment]
@sjakobi: What command are you using to initialize the Nix setup?  `nix-shell` inside of the `./dhall` subdirectory should only be pulling in GHC 8.4.3.  20 GB seems unexpectedly large

**Comment by sjakobi** [comment]
Maybe that's the issue. I was using `nix-build` earlier. I'm retrying with `nix-shell` now (after cleaning my `/nix/store`).

**Comment by sjakobi** [comment]
OK. I just really shouldn't run `nix-build`! ;)

When I ran `nix-shell` on a clean `master` branch, it was super quick, apparently not even pulling in `ghc` itself. My `/nix/store` grew by < 1 GB.

I then switched to my branch for https://github.com/dhall-lang/dhall-haskell/pull/1822 and ran `nix-shell` again. This took considerably longer, on the order of 30 minutes. Regarding storage size, this line from the logs seems pretty accurate:

```
these paths will be fetched (0.00 MiB download, 4244.25 MiB unpacked)
```

(Although this is followed by a bunch of downloads from cache.dhall-lang.org)

Also, unlike `nix-build`, this pulled in only GHC 8.4.3, not 8.6.1 or GHCJS.

Thanks @Gabriel439! I think I'll remember to stick to  `nix-shell` from now on! ;)
